torch/csrc/jit/passes/frozen_conv_folding.cpp

```diff
@@ -79,7 +79,20 @@ void FoldFrozenConvBatchnorm(Block* b) {
       // implementation taken from torch/nn/utils/fusion.py
       Tensor conv_b;
       if (conv->namedInput("bias")->type() == NoneType::get()) {
-        conv_b = at::zeros_like(bn_rm);
+        // If this is on GPU and bias is none and weight was half/bfloat, but
+        // bn_rm was float, then probably this was a case where autocasting
+        // casted inputs to conv. And since CUDA conv implementation requires
+        // all the inputs to have the same scalar dtype, we need to make this
+        // placeholder have the same type as conv_w.
+        at::ScalarType bias_dtype = bn_rm.scalar_type();
+        at::ScalarType weight_dtype = conv_w.scalar_type();
+        at::DeviceType weight_device = conv_w.device().type();
+        if (weight_device == at::kCUDA &&
```

Comment:
is this any reason to not just always put `conv_b` as the same dtype as `weight_dtype` ? for cpu as well ?